File: Frontend/dgu_laf/lib/model/classroom.dart
class Classroom {
  final int classroomId;
  final String buildingName;
  final String roomNumber;

  Classroom({
    required this.classroomId,
    required this.buildingName,
    required this.roomNumber,
  });

  factory Classroom.fromJson(Map<String, dynamic> json) {
    return Classroom(
      classroomId: json['classroom_id'],
      buildingName: json['building_name'],
      roomNumber: json['room_number'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'classroom_id': classroomId,
      'building_name': buildingName,
      'room_number': roomNumber,
    };
  }
}

File: Backend/dgulaf/src/main/java/com/dgulaf/dgulaf/model/Classroom.java
package com.dgulaf.dgulaf.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "classrooms")
public class Classroom {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer classroomId;

    @Column(nullable = false, length = 100)
    private String buildingName;

    @Column(nullable = false, length = 20)
    private String roomNumber;
}
